src/main/java/org/toradocu/extractor/Tag.java

```diff
@@ -1,56 +1,159 @@
 package org.toradocu.extractor;
 
-import java.util.Optional;
+import java.util.Objects;
+import org.toradocu.translator.spec.Specification;
+import org.toradocu.util.Checks;
 
-public interface Tag {
+/** Represents a Javadoc tag. Supported tags are registered in {@code Tag.Kind}. */
```

Comment:
Explain the concept that this abstraction represents.  Does it include the comment or just the tag itself?

Also, this seems to only represent block tags.  Explain that.

The `@code` for Tag.Kind should at least be `@link` to help the reader find the documentation.